### PR TITLE
Feature/agent

### PR DESCRIPTION
**Pull request by trasher**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

### CODE REVIEW

File: inc/snmpcredential.class.php [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+         case 1:
+         case 3:
+            return (string)$this->fields['id'];
+         case 2:
+            return '2c';
+         default:
+            return '';
+
+      }
+   }
```
Comment (cedric-anne) [inline inc/snmpcredential.class.php:108]: Why not relying on `snmpversion` ?

File: inc/snmpcredential.class.php [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+      switch ($this->fields['authentication']) {
+         case 1:
+            return 'MD5';
+         case 2:
+            return 'SHA';
+         default:
+            return '';
+      }
+      return '';
+   }
```
Comment (cedric-anne) [inline inc/snmpcredential.class.php:125]: `authentication` is a varchar and values proposed in form are already `MD5`, `SHA`, ...
I guess this method is obsolete.

File: inc/snmpcredential.class.php [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+         case 1:
+            return 'DES';
+         case 2:
+            return 'AES';
+         case 5:
+            return '3DES';
+         default:
+            return '';
+      }
+   }
```
Comment (cedric-anne) [inline inc/snmpcredential.class.php:143]: `encryption` is a varchar and values proposed in form are already `DES`, `AES`, ...
I guess this method is obsolete.

File: templates/components/form/snmpcredential.html.twig [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+{% extends "generic_show_form.html.twig" %}
+{% import 'components/form/fields_macros.html.twig' as fields %}
+{% set params  = params ?? [] %}
+
+{% block more_fields %}
+   {{ fields.dropdownArrayField(
+      'snmpversion',
+      item.fields['snmpversion'],
+      {0: '-----', '1': '1', '2c': '2c', '3': '3'},
+      __('SNMP version', 'glpiinventory'),
```
Comment (cedric-anne) [inline templates/components/form/snmpcredential.html.twig:41]:
```suggestion
      __('SNMP version'),
```

File: templates/components/form/snmpcredential.html.twig [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+   ) }}
+   {{ fields.htmlField(
+      '',
+      '<h3 class="card-header">v 3</h3>',
+   ) }}
+
+   {{ fields.textField(
+      'community',
+      item.fields['community'],
+      __('Community', 'glpiinventory'),
```
Comment (cedric-anne) [inline templates/components/form/snmpcredential.html.twig:56]:
```suggestion
      __('Community'),
```

File: templates/components/form/snmpcredential.html.twig [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+      _n('User', 'Users', get_plural_number()),
+   ) }}
+
+   {{ fields.nullField() }}
+
+   {{ fields.dropdownArrayField(
+      'authentication',
+      item.fields['authentication'],
+      {0: '-----', 'MD5': 'MD5', 'SHA': 'SHA'},
+      __('Encryption protocol for authentication ', 'glpiinventory'),
```
Comment (cedric-anne) [inline templates/components/form/snmpcredential.html.twig:71]:
```suggestion
      __('Encryption protocol for authentication '),
```

File: templates/components/form/snmpcredential.html.twig [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+      __('Password'),
+   ) }}
+
+   {{ fields.nullField() }}
+
+   {{ fields.dropdownArrayField(
+      'encryption',
+      item.fields['encryption'],
+      {0: '-----', 'DES': 'DES', 'AES128': 'AES128', 'Triple-DES': 'Triple-DES'},
+      __('Encryption protocol for data', 'glpiinventory'),
```
Comment (cedric-anne) [inline templates/components/form/snmpcredential.html.twig:88]:
```suggestion
      __('Encryption protocol for data'),
```

File: inc/snmpcredential.class.php (hunk 2 above)
Comment (trasher) [reply inc/snmpcredential.class.php:125]: Well... The template is wrong, even if type in database was varchar, it did store integers (same for encryption field).

File: inc/snmpcredential.class.php [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+   /**
+    * Real version of SNMP
+    *
+    * @return string
+    */
+   public function getRealVersion(): string {
+      switch ($this->fields['id']) {
+         case 1:
+         case 3:
+            return (string)$this->fields['id'];
```
Comment (trasher) [inline inc/snmpcredential.class.php:101]:
```suggestion
      switch ($this->fields['snmpversion']) {
         case 1:
         case 3:
            return (string)$this->fields['snmpversion'];
```

File: inc/snmpcredential.class.php (hunk 1 above)
Comment (trasher) [reply inc/snmpcredential.class.php]: Indeed, I fixed

File: inc/snmpcredential.class.php (hunk 2 above)
Comment (trasher) [reply inc/snmpcredential.class.php:125]: Template has been fixed

File: inc/snmpcredential.class.php (hunk 3 above)
Comment (trasher) [reply inc/snmpcredential.class.php:143]: Template has been fixed

**Review by cedric-anne — APPROVED** [review]

File: inc/snmpcredential.class.php [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+         case 5:
+            return '3DES';
+         default:
+            return '';
+      }
+   }
+
+   protected function prepareInputs(array $input): array {
+      if (isset($input['auth_passphrase'])) {
+         $input['auth_passphrase'] = Toolbox::sodiumEncrypt($input['auth_passphrase']);
```
Comment (orthagh) [inline inc/snmpcredential.class.php:147]: Note, encrypt/decrypt function will be deprecated soon.
This is just a reminder we should think about change theses line in #9729 after merge of the current pr

**Review by orthagh — APPROVED** [review]

**Pull request merged**